packages/node/src/localEvaluation.ts

```diff
@@ -0,0 +1,184 @@
+   * @param flags The flags to evaluate with the user. If empty, all flags from
+   * the flag cache are evaluated.
+   * @returns The evaluated variants
+   */
+  public async evaluate(
+    user: ExperimentUser,
+    flags?: string[],
+  ): Promise<Variants> {
+    // Evaluate the flag configs and user.
+    const flagConfigs = await this.getFlagConfigs(flags);
+    this.debug('[Experiment] evaluate - user:', user, 'flags:', flags);
+    const resultsString = evaluation.evaluate(
+      JSON.stringify(flagConfigs),
+      JSON.stringify(user),
+    );
+    this.debug('[Experiment] evaluate - result:', resultsString);
+    // Parse variant results
+    const results: EvaluationResult = JSON.parse(resultsString);
+    const variants: Variants = {};
+    Object.keys(results).forEach((key) => {
+      variants[key] = {
+        value: results[key].variant.key,
+        payload: results[key].variant.payload,
+      };
+    });
+    return variants;
+  }
+
+  /**
+   * Start polling for flag configurations.
+   *
+   * You must call this function to begin polling for flag config updates.
+   * The promise returned by this function is resolved when the initial call
+   * to fetch the flag configuration completes.
+   *
+   * Calling this function while the poller is already running will stop and
+   * restart the poller.
+   */
+  public async start(): Promise<void> {
+    this.stop();
+    this.debug('[Experiment] poller - start');
+    this.flagConfigPoller = setInterval(async () => {
+      await this.updateFlagConfigs();
+    }, this.config.flagConfigPollingInterval);
+    if (!this.flagConfigPromise) {
+      this.flagConfigPromise = this.updateFlagConfigs({
+        attempts: 5,
+        min: 1,
+        max: 1,
+        scalar: 1,
+      });
+      await this.flagConfigPromise;
+    }
+  }
+
+  /**
+   * Stop polling for flag configurations.
+   *
+   * Calling this function while the poller is not running will do nothing.
+   */
+  public stop(): void {
+    if (this.flagConfigPoller) {
+      this.debug('[Experiment] poller - stop');
+      clearTimeout(this.flagConfigPoller);
+      this.flagConfigPoller = undefined;
+    }
+  }
+
+  private async fetchFlagConfigs(): Promise<Record<string, string>> {
+    const flagConfigs = await this.doFlagConfigs();
+    return this.parseFlagConfigs(flagConfigs);
+  }
+
+  private async getFlagConfigs(flagKeys?: string[]): Promise<string[]> {
+    if (this.flagConfigPromise) {
+      this.debug('[Experiment] waiting for flag configs');
+      await this.flagConfigPromise;
+    }
+    return Object.values(this.flagConfigCache.get(flagKeys));
+  }
+
+  private async updateFlagConfigs(
+    backoffPolicy?: BackoffPolicy,
+  ): Promise<void> {
+    this.debug('[Experiment] updating flag configs');
+    return await doWithBackoff<void>(async () => {
+      const flagConfigs = await this.fetchFlagConfigs();
+      this.flagConfigCache.clear();
+      this.flagConfigCache.put(flagConfigs);
```

Comment:
optimization: `JSON.stringify()` here and you won't need to stringify everytime evaluate is called.

Reply:
Stringifying here wont avoid another stringify because the evaluate function allows the user to evaluate a subset of flags from the whole. E.g. if we fetch flags A, B, and C, and the user only wants to evaluate B and C, then we need a JSON array of flag configs B and C. Since we store the flags in a map in the cache, we need to access flag configs B and C from the map, put them into an array, and then stringify the array. to pass to the evaluation engine.

Reply:
Ah, my bad. I really meant stringifying in the flagConfigCache.put method. The typing of cache being `Record<string, string>` made me think that it was intended to store FlagConfigs as a string.

Reply:
This was my mistake originally, The type was string but really it was storing an object, everything worked fine....JavaScript 🤷‍♂️